lib/symboldatabase.h

```diff
@@ -52,7 +52,7 @@ class ValueType;
 /**
  * @brief Access control enumerations.
  */
-enum class AccessControl { Public, Protected, Private, Global, Namespace, Argument, Local, Throw };
+enum class AccessControl : uint8_t { Public, Protected, Private, Global, Namespace, Argument, Local, Throw };
```

Comment:
I am not sure if this helps so much. there's risk the data will be padded to 32-bit boundaries anyway right?

Reply:
No, a type of 1 byte is usually padded on 1 byte boundaries. That means that for reducing memory usage variables of small types should be groupped. You can check this with sizeof().

Reply:
> That means that for reducing memory usage variables of small types should be groupped.

yes but my assumption was that we don't group members by size and we don't have much opportunities to do it neither.

Reply:
I just don't know if all enum types with a limited number of constants should always be some uint8_t .. I'd use uint8_t if I want to use the the enum constant values..

Reply:
I did reorder several members (and I am not the first one doing this in cppcheck, I remember commits by other developers in the past on this topic).

Reply:
We could limit this to those enums that are used as member variables somewhere.

Reply:
You can also use `-Wpadded` to see how much padding is actually being performed. II am also trying to get this warning enabled in the CI at least.

CLion also shows information about the object size and padding used. I assume it simply takes it from clangd so if you use that you also get information on it.